### PR TITLE
feat(service-portal): vehicle additions

### PR DESCRIPTION
**Pull request by thordurhhh**
## What

Additions and updates to the vehicle api and ui.
This feature is behind a feature-flag.

## Why

API updated. More information added.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by Valur — APPROVED** [review]

File: libs/service-portal/vehicles/src/components/DetailTable/InspectionInfoItem.tsx [hunk 1]
```diff
@@ -61,12 +66,26 @@ const InspectionInfoItem = ({ data }: PropTypes) => {
...
+              ? amountFormat(data.inspectionFine)
+              : ''
+          }
+        />
       </Row>
-      <Row>
+      {/* <Row>
         <Column label={messages.plateLocation} value={null} />
-      </Row>
+      </Row> */}
```
Comment (saevarma) [inline libs/service-portal/vehicles/src/components/DetailTable/InspectionInfoItem.tsx:88]: Remove redundant comment?

File: libs/service-portal/vehicles/src/screens/VehicleDetail/VehicleDetail.tsx [hunk 2]
```diff
@@ -106,19 +106,31 @@ const VehicleDetail: ServicePortalModuleComponent = ({ userInfo }) => {
 
         <UserInfoLine
           label={formatMessage(messages.capacity)}
-          content={mainInfo?.cubicCapacity?.toString() + ' cc.'}
+          content={
+            mainInfo?.cubicCapacity
+              ? mainInfo?.cubicCapacity?.toString() + ' cc.'
```
Comment (saevarma) [inline libs/service-portal/vehicles/src/screens/VehicleDetail/VehicleDetail.tsx:111]: You are already checking if the property is set so optional chaining is unnecessary.
```suggestion
              ? mainInfo.cubicCapacity.toString() + ' cc.'
```

File: libs/service-portal/vehicles/src/screens/VehicleDetail/VehicleDetail.tsx [hunk 3]
```diff
@@ -106,19 +106,31 @@ const VehicleDetail: ServicePortalModuleComponent = ({ userInfo }) => {
...
+          }
           loading={loading}
         />
         <Divider />
         <UserInfoLine
           label={formatMessage(messages.trailerWithBrakes)}
-          content={mainInfo?.trailerWithBrakesWeight?.toString() + ' kg.'}
+          content={
+            mainInfo?.trailerWithBrakesWeight
+              ? mainInfo?.trailerWithBrakesWeight?.toString() + ' kg.'
```
Comment (saevarma) [inline libs/service-portal/vehicles/src/screens/VehicleDetail/VehicleDetail.tsx:121]: Same about unnecessary optional chaining

File: libs/service-portal/vehicles/src/screens/VehicleDetail/VehicleDetail.tsx [hunk 4]
```diff
@@ -106,19 +106,31 @@ const VehicleDetail: ServicePortalModuleComponent = ({ userInfo }) => {
...
+          }
           loading={loading}
         />
         <Divider />
         <UserInfoLine
           label={formatMessage(messages.trailerWithoutBrakes)}
-          content={mainInfo?.trailerWithoutBrakesWeight?.toString() + ' kg.'}
+          content={
+            mainInfo?.trailerWithoutBrakesWeight
+              ? mainInfo?.trailerWithoutBrakesWeight?.toString() + ' kg.'
```
Comment (saevarma) [inline libs/service-portal/vehicles/src/screens/VehicleDetail/VehicleDetail.tsx:131]: Same about unnecessary optional chaining

**Review by saevarma — CHANGES_REQUESTED** [review]

File: libs/api/domains/vehicles/src/lib/api-domains-vehicles.service.ts [hunk 5]
```diff
@@ -188,13 +193,25 @@ export class VehiclesService {
           carryingCapacity: data.techincal?.mass?.masscapacity,
           axleTotalWeight: axleMaxWeight,
           axle: axles,
+          tyres: {
+            tyre1: data.techincal?.tyre?.tyreaxle1,
```
[the remark this answers is not in the document]
Comment (thordurhhh) [reply libs/api/domains/vehicles/src/lib/api-domains-vehicles.service.ts:197]: The naming of the properties in the API is a bit strange, so this is actually referring to
> A tyre on axle nr 1

The axle details are here then: `axle: axles`

File: libs/api/domains/vehicles/src/models/getVehicleDetail.model.ts [hunk 6]
```diff
@@ -142,6 +160,18 @@ export class VehiclesInspectionInfo {
...
+  insuranceStatus?: boolean | null // insurancestatus
+
+  @Field(() => Number, { nullable: true })
+  encumbrances?: number | null // hasEncumbrances
+
+  @Field(() => Number, { nullable: true })
+  carTax?: number | null // bifreidagjald
+
+  @Field(() => Number, { nullable: true })
+  inspectionFine?: number | null // inspectionfine
```
[the remark this answers is not in the document]
Comment (thordurhhh) [reply libs/api/domains/vehicles/src/models/getVehicleDetail.model.ts:174]: Yeah, this should probably be mortgages. It makes more sense.

**Comment by thordurhhh** [comment]
Great, thanks @saevarma .

These should all be addressed now 👍

**Review by saevarma — APPROVED** [review]
Looks good just two minor comments I missed before.
I removed the `automerge` label, so I could approve and you can add it after you address those two comments :)

**Pull request merged**